File: Sieve_of_Eratosthenes/Ruby/mitogh/sieve.rb
def sieve(n) 
  primes = Array.new(n, true)
  limit = Math.sqrt(n).floor
  primes[0], primes[1] = nil, nil

  for i in 2..limit
    if primes[i]
      j, k = 0, 0
      while( j <= n ) do
        j = i ** 2 + (k * i)
        k += 1
        primes[j] = nil
      end
    end
  end
  primes
end

File: Sieve_of_Eratosthenes/PHP/mitogh/sieve.php
<?php
function sieve($n = 0){
    $numbers = array_fill(2, $n + 1, true);
    $numbers[0] = false;
    $numbers[1] = false;

    $limit = floor(sqrt($n));

    for($i = 2; $i <= $limit; $i++){
        if($numbers[$i]){
            while(true){
                $j = pow($i,2) + ($k++ * $i);
                if($j <= $n){
                    $numbers[$j] = false;
                }else{
                    break;
                }
            }
        }
    }
    return $numbers;
}
